### PR TITLE
Sharing button for new Assistant Builder

### LINKED ISSUE #3780: [Meru - AB] Layout - Top-right sharing button
Details on what to do in the **[Figma](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4900-40669&mode=design&t=6tNvt0C7wUTsOHc4-0)** (also, see how it works in the [Prototype](https://www.figma.com/proto/QOxHwppG64GtPocpDhS6Y7/Product?page-id=3366%3A31345&type=design&node-id=4900-40669&viewport=1173%2C2546%2C0.27&t=lV2M2VcNHLWO3Bpe-1&scaling=min-zoom&starting-point-node-id=4900%3A40669) + context & discussion in the [making issue](https://github.com/dust-tt/decisions/issues/164))

Note the various [dropdown variants](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4743-20569&mode=design&t=6tNvt0C7wUTsOHc4-4)

Clicking on the button opens the dropdown. Wwithin it, clicking on the chip opens the existing sharing dropdown (see Sharing.tsx)

### PR DESCRIPTION
**Pull request by spolu**
## Description

Fixes: https://github.com/dust-tt/dust/issues/3780

Add a Sharing Button component as part of Meru new Assistant Builder.

Screenshot from 2024-02-21 18-26-38
Screenshot from 2024-02-21 18-26-32
Screenshot from 2024-02-21 18-26-23
Screenshot from 2024-02-21 18-26-13

We still have a popup for confirmation when changing here.

## Risk

N/A

## Deploy Plan

- deploy `front`

### CODE REVIEW

**Comment by spolu** [comment]
Note: will require an update once ShapesShareIcon is made available (cc @Duncid)

File: front/components/assistant/Sharing.tsx [hunk 1]
```diff
@@ -114,34 +116,50 @@ export function SharingSection({
   });
   const assistantName = agentConfiguration?.name;
 
+  console.log(agentUsage);
```
Comment (philipperolet) [inline front/components/assistant/Sharing.tsx:119]:
```suggestion
```

File: front/lib/assistant.ts [hunk 2]
```diff
@@ -113,7 +113,7 @@ export function assistantUsageMessage({
   isError,
   shortVersion,
 }: {
-  assistantName: string;
+  assistantName: string | null;
```
Comment (philipperolet) [inline front/lib/assistant.ts:116]: Why accept null ? This message should not show for assistants not yet created, and for those created we have the name. Refusing null is guarding us from using the message when we shouldn't no?

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 3]
```diff
@@ -92,6 +93,8 @@ import { isUpgraded } from "@app/lib/plans/plan_codes";
 import { useSlackChannelsLinkedWithAgent, useUser } from "@app/lib/swr";
 import { classNames } from "@app/lib/utils";
 
+import { SharingButton } from "../assistant/Sharing";
```
Comment (philipperolet) [inline front/components/assistant_builder/AssistantBuilder.tsx:96]: nit: relative import

File: front/components/assistant/Sharing.tsx [hunk 4]
```diff
@@ -114,34 +116,50 @@ export function SharingSection({
...
+            initialScope={initialScope}
+            newScope={newScope}
+            setNewScope={setNewScope}
+          />
+          <div className="text-sm text-element-700">
+            <div>
+              {SCOPE_INFO[newScope].text}{" "}
+              {agentUsage &&
+              agentUsage.userCount &&
+              agentUsage.userCount > 0 &&
```
Comment (philipperolet) [inline front/components/assistant/Sharing.tsx:154]: here the best IMHO would be to replace with `agentUsage.usersWithAgentInListCount` introduced ~ a week ago => fits the original framing intent in Ahuna 2, +  text does not show when creating (at a time when it does not make sense), etc.

**Review by philipperolet — CHANGES_REQUESTED** [review]
LGTM :+1: 2 nits, and 2 coms on agentUsage. Good for me to merge without re-review once those are adressed as you see fit

File: front/lib/assistant.ts (hunk 2 above)
Comment (spolu) [reply front/lib/assistant.ts:116]: The goal is to match the design where we don't want to repeat the assistant name in the assistant builder and use "This assistant" instead

File: front/components/assistant_builder/AssistantBuilder.tsx (hunk 3 above)
Comment (spolu) [reply front/components/assistant_builder/AssistantBuilder.tsx:96]: :+1:

**Pull request merged**